src/invision/components/movie-list/movie-list.js

```diff
@@ -1,90 +1,133 @@
-import React from 'react';
+import React, { useState, useCallback } from 'react';
 import thumbnail from './thumbnail.png';
 import MovieItem from '../movie-item/movie-item'
 import './movie-list.scss';
 import MovieService from '../../services/movie.service';
 
-export default class MovieList extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      movieList: [],
-      sortValue: 'RELEASE DATE',
-    };
+import MoviePopup from '../movie-popup/movie.popup';
+import DeletePopup from '../delete-movie/delete-popup';
+import useFetchMovie from './fetch-movie.hook';
 
-    this.getMovieList();
+export default function MovieList() {
 
-    this.sort = this.sort.bind(this);
-  }
+  const [movieState, updateMovieState] = useState({
+    movieList: [],
+    sortValue: 'RELEASE DATE',
+    isEditMovieVisible: false,
+    isDeleteMovieVisible: false,
+    selectedMovie: null
+  })
 
-  getMovieList() {
+  useFetchMovie(() => {
     new MovieService().getMovieList().then(movieList => {
```

Comment:
The main idea of the custom hooks, that you can hide and generalize some functionality. It means, that yo should move **MovieService** to the custom hook or name your custom hook like **componentDidMount**